### DISCUSSION TITLE
Stacked/overlaping masks are XORED but detections boxes are not (mask should not be).

### DISCUSSION THREAD

**Discussion by ExtReMLapin** [ultralytics/ultralytics#7878] [Q&A] [answered]
-snip- ignore

**Comment by ExtReMLapin** [comment]
It doesn't seems to be caused by NMS as NMS isn't perfect but when nutturing NMS, it doesn't generate any of the expected masks

**Reply by pderrenger** [reply]
@ExtReMLapin it sounds like you're experiencing an issue where the masks are being XORed (exclusive OR operation) when they overlap, but the detection boxes are not. This could be due to the way masks are being processed or combined post-detection.

To better assist you, could you please provide more details or share a snippet of your code? This will help in understanding the exact problem and suggesting a potential solution. Meanwhile, ensure that your post-processing steps for mask combination are correctly configured to handle overlapping instances as desired. 🛠️

If you believe this is a bug or a feature request, consider opening an issue on the Ultralytics YOLOv8 GitHub repository with a detailed explanation and any relevant code or images to reproduce the issue. The development team and community can then collaborate to address it. 👍

**Comment by ExtReMLapin** [comment, marked as answer]
Not possible with YOLO https://github.com/ultralytics/ultralytics/issues/3213

**Reply by glenn-jocher** [reply]
Thanks for the issue link!
